Sources/ArgumentParser/Completions/BashCompletionsGenerator.swift

```diff
@@ -122,14 +122,16 @@ struct BashCompletionsGenerator {
 
   /// Returns the option and flag names that can be top-level completions.
   fileprivate static func generateArgumentWords(_ commands: [ParsableCommand.Type]) -> [String] {
-    commands.argumentsForHelp().flatMap { $0.bashCompletionWords() }
+    commands
+      .argumentsForHelp(minimumVisibility: .default)
+      .flatMap { $0.bashCompletionWords() }
   }
 
   /// Returns additional top-level completions from positional arguments.
   ///
   /// These consist of completions that are defined as `.list` or `.custom`.
   fileprivate static func generateArgumentCompletions(_ commands: [ParsableCommand.Type]) -> [String] {
-    ArgumentSet(commands.last!)
+    ArgumentSet(commands.last!, minimumVisibility: .default)
```

Comment:
What do you think about this? With the visibility approach, "hidden" just means "display on demand", so maybe they should appear in completions?

Reply:
I'm not sure which way to go tbh, on one hand it would be nice to get completions, on the other I could imagine SPM for example not wanting to encourage to use `-Xfrontend` and similar features.

I'm happy to go either way

Reply:
I just realized visibility hasn't been plumbed into ArgumentDefinition yet. I think its a bigger PR to support hidden completions properly, I think we should push it to another PR.